src/robot/running/handlers.py

```diff
@@ -382,3 +382,26 @@ def _parse_arguments(self, handler_method):
         parser = JavaArgumentParser(type='Test Library')
         signatures = self._get_signatures(handler_method)
         return parser.parse(signatures, self.library.name)
+
+
+class EmbeddedArgsTemplate(object):
+
+    def __init__(self, embedded, orig_handler):
+        self.orig_handler = orig_handler
+        self._orig_run = orig_handler._run
+        self.orig_handler._run = self._run
```

Comment:
This is a bit ugly but also very clever. A cleaner solution would likely require a bigger refactoring and I don't think that's worth the effort.

Only thing I'm a little worried about is that is there a change that the original handler would be used somehow without embedded arguments. In that case the original handler should use its normal `run` and `_run`. In practice it's very unlikely that would cause problems with existing keywords because normal keywords cannot contain `${}` in their name. Just deciding that such usage is not supported would probably be OK.